ballista/executor/src/circuit_breaker/client.rs

```diff
@@ -0,0 +1,245 @@
+use anyhow::Error;
+use ballista_core::{
+    error::BallistaError,
+    serde::protobuf::{self, CircuitBreakerUpdateRequest},
+};
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio_stream::{wrappers::ReceiverStream, StreamExt};
+use tracing::{info, warn};
+
+use crate::{
+    circuit_breaker::stream::CircuitBreakerUpdate,
+    scheduler_client_registry::SchedulerClientRegistry,
+};
+
+pub struct CircuitBreakerClient {
```

Comment:
I'm not wild about passing around the `task_id` as a string and parsing it to try and extract the information we need. Seems like we could just add a strongly typed `TaskId` here and inject it when we construct this guy and add it to the `SessionConfig`

Reply:
The reason I did it this way is because it should be the responsibility of the scheduler to correlate tasks with jobs, not the executor, as the executor is clearly designed to not know anything other than the task ID.

But let me check if I can maybe correlate in a different way.